pkg/policy/rule.go

```diff
@@ -35,19 +36,24 @@ func (r *rule) String() string {
 // port and protocol with the contents of the provided PortRule. If the rule
 // being merged has conflicting L7 rules with those already in the provided
 // L4PolicyMap for the specified port-protocol tuple, it returns an error.
-func mergeL4IngressPort(ctx *SearchContext, endpoints []api.EndpointSelector, r api.PortRule, p api.PortProtocol,
+//
+// If any rules contain L7 rules that overlap with the endpointsWithL3Override,
+// then for the endpoints with L3 override, the L7 rules will be translated
+// into L7 wildcards (ie, traffic will be forwarded to the proxy for endpoints
+// matching those labels, but the proxy will allow all such traffic).
+func mergeL4IngressPort(ctx *SearchContext, endpoints []api.EndpointSelector, endpointsWithL3Override []labels.LabelArray, r api.PortRule, p api.PortProtocol,
```

Comment:
Is this only necessary for ingress ? I'm guessing that it is only necessary for ingress based on the documentation for the CLI option for "always allow localhost":

https://github.com/cilium/cilium/blob/ad437706dd4ee26d704e05e3560d74ae7b2c113c/daemon/main.go#L310-L311

Reply:
Yes, it's only for ingress.